### PR TITLE
folder_block_ops: refactor file data logic into its own struct

### PR DESCRIPTION
**Pull request by strib**
This will make it easier to add support for multiple levels of
indirection in a future PR.  Note that this just moves code around, it
shouldn't change behavior at all.

Issue: KBFS-35

### CODE REVIEW

**Comment by strib** [comment]
(Adding @jzila as a reviewer to get more up to speed with block ops stuff.)

File: libkbfs/file_data.go [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+	path, blockReqType) (*FileBlock, error)
+type dirtyBlockCacher func(ptr BlockPointer, block Block) error
+
+// fileData is a helper struct for accessing and manipulating data
+// within a file.  It's meant for use within a single scope, not for
+// long-term storage.  The caller must ensure goroutine-safety.
+type fileData struct {
+	file   path
+	uid    keybase1.UID
+	crypto Crypto
```
Comment (akalin-keybase) [inline libkbfs/file_data.go]: CryptoCommon instead?

File: libkbfs/file_data.go [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+		}
+	}
+
+	return ptr, parentBlock, indexInParent, block, nextBlockStartOff,
+		startOff, nil
+}
+
+// createIndirectBlock creates a new indirect block and pick a new id
+// for the existing block, and use the existing block's ID for the new
+// indirect block that becomes the parent.
```
Comment (strib) [inline libkbfs/file_data.go]: The correct IDs are filled in during `folderBlockOps.startSyncWrite()`, after all of the child blocks have been passed through `BlockOps.Ready()`.

https://github.com/keybase/kbfs/blob/aaf87b909302e7c5738b2ddb9207e60b8d148ac1/libkbfs/folder_block_ops.go#L2217

**Comment by strib** [comment]
Filed KBFS-1715 for the test flake, definitely seems unrelated to this PR.

**Pull request merged**